### PR TITLE
Prefer std::vector<T, vespalib::allocator_large> over vespalib::Array…

### PR DESCRIPTION
**Pull request by baldersheim**
…<T> step 1.

@vekterli or @havardpe PR

### CODE REVIEW

**Review by havardpe — APPROVED** [review]

File: searchlib/src/vespa/searchlib/attribute/enum_store_types.h [hunk 1]
```diff
@@ -4,14 +4,14 @@
...
 #include <vespa/vespalib/datastore/entryref.h>
-#include <vespa/vespalib/util/array.h>
+#include <vespa/vespalib/stllike/allocator.h>
 
 namespace search::enumstore {
 
 using Index = vespalib::datastore::EntryRef;
 using InternalIndex = vespalib::datastore::EntryRefT<22>;
-using IndexVector = vespalib::Array<Index>;
+using IndexVector = std::vector<Index, vespalib::allocator_large<Index>>;
```
Comment (vekterli) [inline searchlib/src/vespa/searchlib/attribute/enum_store_types.h:13]: Should `<vector>` be added as an explicit include? Currently transitively included via `iattributevector.h`, but this is a bit opaque

File: searchlib/src/vespa/searchlib/grouping/groupengine.h [hunk 2]
```diff
@@ -123,15 +123,15 @@ class GroupEngine : protected Collect
     size_t getIdBase(GroupRef g) const { return _idByteSize*g; }
 
     using IdList = std::unique_ptr<expression::ResultNodeVector>;
-    typedef vespalib::Array<Children *> GroupBacking;
-    typedef std::vector<double> RankV;
-    typedef vespalib::Array<uint8_t> IdBacking;
+    using GroupBacking = std::vector<Children *>;
+    using RankV = std::vector<double>;
+    using IdBacking = std::vector<uint8_t>;
```
Comment (vekterli) [inline searchlib/src/vespa/searchlib/grouping/groupengine.h:128]: Should these vectors use `allocator_large` as well, or is it fine to just put things on the heap?

File: vespalib/src/vespa/vespalib/stllike/hashtable.h [hunk 3]
```diff
@@ -184,7 +185,7 @@ class hashtable : public hashtable_base
 private:
     using Node=hash_node<Value>;
 protected:
-    using NodeStore = vespalib::Array<Node>;
+    using NodeStore = std::vector<Node, allocator_large<Node>>;
```
Comment (vekterli) [inline vespalib/src/vespa/vespalib/stllike/hashtable.h:188]: Consider explicitly including `<vector>` in this file as well

**Review by vekterli — APPROVED** [review]
:+1:

File: searchlib/src/vespa/searchlib/attribute/enum_store_types.h (hunk 1 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/attribute/enum_store_types.h:13]: There will be follow ups cleaning up includes a bit more. Will probably happen then.

File: vespalib/src/vespa/vespalib/stllike/hashtable.h (hunk 3 above)
Comment (baldersheim) [reply vespalib/src/vespa/vespalib/stllike/hashtable.h:188]: There will be follow ups cleaning up includes a bit more. Will probably happen then.

File: searchlib/src/vespa/searchlib/grouping/groupengine.h (hunk 2 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/grouping/groupengine.h:128]: Yes, these lists will should not be so large that direct mmapping is required. Then there will be other issues.

**Pull request merged**